### DISCUSSION TITLE
pandoc for software documentation, like antora

### DISCUSSION THREAD

**Discussion by RudolfVonKrugstein** [jgm/pandoc#10272] [Q&A] [answered]
Does a tool like antora for asciidoc, exist for pandoc?

By "a tool like antora" I mean a tool that renders multiple markdown files onto an HTML structure, with an index in a sidebar allowing me to easily navigate it in the browser?

**Comment by jgm** [comment, marked as answer]
Well, maybe not exactly what you want.  You can specify multiple markdown files, and pandoc will construct one big document out of them; if you use `-t chunkedhtml`, you'll get something like you're describing (but probably not exactly). Try it.

**Reply by RudolfVonKrugstein** [reply]
Thank your for the tip! I want something that works for me, and don't know 100% yet what that means :). I will try it.

**Reply by bpj** [reply]
Maybe something as simple as a Makefile which creates the directory structure and converts each MD file with Pandoc will do.

 I used to use a Perl script which walked a directory structure with MD files and built a parallel directory structure with HTML/PDF by shelling out to pandoc as it went (with any images in a separate img directory which was symlinked during development, then copied). The last (IIRC) version even extracted the title etc. from a YAML block at the top of each file and built an index for each directory using IPC::Run3, but I found that I preferred writing/maintaining the index files by hand. 

Now something like that might be possible to do with Lua. The tricky bit would be to get the directory listings without using `ls`! I would probably still get the directory listings through a `pandoc.pipe` to a Perl script which does the actual directory walking. Or run the Perl script first and have it create the directories and write a JSON file with the paths which `pandoc-lua` could read.

**Comment by tarleb** [comment]
There is [Quarto](https://quarto.org), and some website generators like Hugo have pandoc plugins.

**Reply by RudolfVonKrugstein** [reply]
Oh, that looks good. Thank you!
